contrib/devtools/symbol-check.py

```diff
@@ -60,42 +60,45 @@
 ALLOW_DYNAMIC_LIBSTDCXX = False
 
 MAX_VERSIONS = {
-    'GCC':   (4,5,0),
-    'GLIBC': (2,23),
+    b'GCC':   (4,5,0),
+    b'GLIBC': (2,23),
 }
 if ALLOW_DYNAMIC_LIBSTDCXX:
     MAX_VERSIONS.update({
-        'GLIBCXX': (3,4,14),
-        'CXXABI':  (1,3,4),
+        b'GLIBCXX': (3,4,14),
+        b'CXXABI':  (1,3,4),
     })
 
 # See here for a description of _IO_stdin_used:
 # https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=634261#109
 
 # Ignore symbols that are exported as part of every executable
 IGNORE_EXPORTS = {
-    '_edata', '_end', '_init', '__bss_start', '_fini', '_IO_stdin_used'
+    b'_edata', b'_end', b'_init', b'__bss_start', b'_fini', b'_IO_stdin_used'
```

Comment:
Upstream migrated this file here: https://github.com/bitcoin/bitcoin/pull/7723/commits/18f05c765c800126b74a6d5b7f33cef7c9aae1b7

I would prefer we backport the upstream PRs https://github.com/bitcoin/bitcoin/pull/7723 and https://github.com/bitcoin/bitcoin/pull/11881 (for the files we haven't yet migrated), along with any PRs blocking those, and then apply any additional changes we want on top.